benchmarks/src/main_config.toml

```diff
@@ -0,0 +1,9 @@
+[heat.cpu]
+code_target = "CPUTarget"
+float_type = ["Float32", "Float64"]
+resolution = [5, 2, 1]
+
+[heat.cuda]
+code_target = "CUDATarget"
+float_type = ["Float32", "Float64"]
+resolution = [5, 2, 1]
```

Comment:
Now that I understand that:

1. each simulation still needs code for its set up that needs to go in a folder
2. the simulations could in theory have different dependencies (because they are in their own folder)
3. there is a generate_configs script that is copying this data into individual config tomls for each simulation.

It makes sense to me that these config tomls actually live 1 per folder and specify all the configurations of the same simulation that you want to run. But the global `main_config.toml` is actually just a list of all the configurations that you want to run. Much like in the docs, `docs/main.jl` has a big list of all the pages in the docs to tell you which markdown files to include.

Reply:
The idea with having different configs per architecture was that a user might want different configurations for each. For example, gpu sims could include larger mesh sizes or longer times. I think this flexibility is nice to have.

I like your idea of the role of the `main_config.toml` since the data processing is the same, we just change where we read the data. Plus, if we keep separate tomls for each architecture then the user can choose to run just cpu benchmarks or just gpu benchmarks.

Just thinking through the pipeline, the user can just provide their `main_config.toml` which lists all sims/arches to run. `main.jl` can then read the main config and auto-run the config generation that looks for the user provided `toml` for each of these sims and generates a list of configs for tasks. It'll then basically just run the benchmarks as it does now. 

We can implement the `make all` feature by scanning the `src` dir for valid physics folders and then running through the same pipeline as above.

Reply:
As for post-processing, the idea now is to collect all the data from the benchmarks into `.jld2` files that can be collected into dataframes using DrWatson. I've provided a basic post-processing step in `scripts/post_processing/default_out.jl` but we can add more scripts for different post-processing as we find the need for it. This could include deciding to output to markdown or HTML or whatever else.

Reply:
I don't see that post processing script. Did you commit it?

Reply:
Yeah that's my bad, I just pushed those.